### DISCUSSION TITLE
Best way to securely store database connection string for Database Connection of Static Web App

### DISCUSSION THREAD

**Discussion by jacksorjacksor** [Azure/static-web-apps#1323] [Q&A] [answered]
Hi! How can we use a configuration/environment file to populate the database connection strings at the CI/CD pipeline step, and thus be able to keep the database connection string secret/not committed to the repo?

This was asked previously by @RayMaAU in [this discussion comment](https://github.com/Azure/static-web-apps/discussions/1111#discussioncomment-6091339).

Thanks!

**Comment by thomasgauvin** [comment, marked as answer]
Hi @jacksorjacksor, when you link your database to your Static Web Apps resource with the database connections feature, you specify your secrets to connect to the database. Only your configuration file, not your secrets, need to be stored in your repository. When we make the connection to your database, we use the configuration file provided in your repository and override it with the secrets collected from the Azure Portal/ARM when you create the database connection.

**Comment by jacksorjacksor** [comment]
Thanks @thomasgauvin for that, we'll be able to utilise this approach during deployment.
Do you have a preferred approach for local development? i.e. Are you setting machine-wide environment variables or is there a method of setting SPA-specific secrets?
